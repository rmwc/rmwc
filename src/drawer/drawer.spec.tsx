import React from 'react';
import { render } from '@testing-library/react';
import {
  Drawer,
  DrawerContent,
  DrawerHeader,
  DrawerTitle,
  DrawerSubtitle,
  DrawerAppContent
} from './';

describe('Drawer', () => {
  it('Drawer renders', () => {
    const { asFragment } = render(
      <Drawer>
        <DrawerHeader>
          <DrawerTitle>Title</DrawerTitle>
          <DrawerSubtitle>Subtitle</DrawerSubtitle>
        </DrawerHeader>
        <DrawerContent />
      </Drawer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('dismissible Drawer renders', () => {
    const { asFragment } = render(
      <React.Fragment>
        <Drawer dismissible>
          <DrawerHeader>
            <DrawerTitle>Title</DrawerTitle>
            <DrawerSubtitle>Subtitle</DrawerSubtitle>
          </DrawerHeader>
          <DrawerContent />
        </Drawer>
        <DrawerAppContent>Test</DrawerAppContent>
      </React.Fragment>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('modal Drawer renders', () => {
    const { asFragment } = render(
      <React.Fragment>
        <Drawer modal>
          <DrawerHeader>
            <DrawerTitle>Title</DrawerTitle>
            <DrawerSubtitle>Subtitle</DrawerSubtitle>
          </DrawerHeader>
          <DrawerContent />
        </Drawer>
      </React.Fragment>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('can open', () => {
    const { container, rerender } = render(
      <Drawer dismissible>
        <DrawerHeader>
          <DrawerTitle>Title</DrawerTitle>
          <DrawerSubtitle>Subtitle</DrawerSubtitle>
        </DrawerHeader>
        <DrawerContent />
      </Drawer>
    );
    expect(container.firstChild).not.toHaveClass('mdc-drawer--open');

    rerender(
      <Drawer dismissible open>
        <DrawerHeader>
          <DrawerTitle>Title</DrawerTitle>
          <DrawerSubtitle>Subtitle</DrawerSubtitle>
        </DrawerHeader>
        <DrawerContent />
      </Drawer>
    );
    expect(container.firstChild).toHaveClass('mdc-drawer--open');
  });

  it('can have custom classnames', () => {
    [Drawer, DrawerHeader, DrawerContent, DrawerTitle, DrawerSubtitle].forEach(
      (Component: React.ComponentType<any>) => {
        const { container } = render(
          <Component className={'my-custom-classname'} />
        );
        expect(container.firstChild).toHaveClass('my-custom-classname');
      }
    );
  });
});
