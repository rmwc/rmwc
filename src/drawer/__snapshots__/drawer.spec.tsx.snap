// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Drawer renders 1`] = `
<DocumentFragment>
  <aside
    class="mdc-drawer"
  >
    <div
      class="mdc-drawer__header"
    >
      <div
        class="mdc-drawer__title"
      >
        Title
      </div>
      <div
        class="mdc-drawer__subtitle"
      >
        Subtitle
      </div>
    </div>
    <div
      class="mdc-drawer__content"
    />
  </aside>
</DocumentFragment>
`;

exports[`Drawer dismissible Drawer renders 1`] = `
<DocumentFragment>
  <aside
    class="mdc-drawer mdc-drawer--dismissible"
  >
    <div
      class="mdc-drawer__header"
    >
      <div
        class="mdc-drawer__title"
      >
        Title
      </div>
      <div
        class="mdc-drawer__subtitle"
      >
        Subtitle
      </div>
    </div>
    <div
      class="mdc-drawer__content"
    />
  </aside>
  <div
    class="mdc-drawer-app-content"
  >
    Test
  </div>
</DocumentFragment>
`;

exports[`Drawer modal Drawer renders 1`] = `
<DocumentFragment>
  <aside
    class="mdc-drawer mdc-drawer--modal"
  >
    <div
      class="mdc-drawer__header"
    >
      <div
        class="mdc-drawer__title"
      >
        Title
      </div>
      <div
        class="mdc-drawer__subtitle"
      >
        Subtitle
      </div>
    </div>
    <div
      class="mdc-drawer__content"
    />
  </aside>
  <div
    class="mdc-drawer-scrim"
  />
</DocumentFragment>
`;
